src/middlewared/middlewared/plugins/vm.py

```diff
@@ -929,13 +929,23 @@ def random_mac(self):
         else:
             return update_devices
 
-    @accepts(Int('id'))
-    async def do_delete(self, id):
+    @accepts(Int('id'), Bool('disks', default=False))
+    async def do_delete(self, id, disks):
         """Delete a VM."""
         status = await self.status(id)
         if isinstance(status, dict):
             if status.get('state') == 'RUNNING':
                 await self.stop(id)
+        if disks:
+            devices = await self.middleware.call('vm.device.query', [
+                ('vm', '=', id), ('dtype', '=', 'DISK')
+            ])
+
+            for device in devices:
+                disk_name = device['attributes']['path'].rsplit(
+                    '/dev/zvol/'
```

Comment:
There is no guarantee it is a zvol, I think. In theory it could be a disk device. Need to make sure its a zvol first.